aptos-move/aptos-vm/src/sharded_block_executor/local_executor_shard.rs

```diff
@@ -145,18 +184,20 @@ impl<S: StateView + Sync + Send + 'static> ExecutorClient<S> for LocalExecutorCl
                 ))
                 .unwrap();
         }
-    }
 
-    fn get_execution_result(&self) -> Result<Vec<Vec<Vec<TransactionOutput>>>, VMStatus> {
-        trace!("LocalExecutorClient Waiting for results");
-        let mut results = vec![];
-        for (i, rx) in self.result_rxs.iter().enumerate() {
-            results.push(
-                rx.recv()
-                    .unwrap_or_else(|_| panic!("Did not receive output from shard {}", i))?,
-            );
-        }
-        Ok(results)
+        // This means that we are executing the global transactions concurrently with the individual shards but the
+        // global transactions will be blocked for cross shard transaction results. This hopefully will help with
+        // finishing the global transactions faster but we need to evaluate if this causes thread contention. If it
+        // does, then we can simply move this call to the end of the function.
```

Comment:
so it's effectively a special shard...

Could it be too early to make a such a formal change to introduce global round/shard, before we are sure it's needed...?

Reply:
@zjma - this will be needed sooner than later. For non-P txns for example, it might be tricky or very inefficient to shard them. So we can put them in a global shard.

Reply:
Also, this is optional, so we can run various tests to see the performance before making this as a default.